### PR TITLE
Add various features

### PR DESCRIPTION
**Pull request by kagikn**

### CODE REVIEW

File: source/scripting/Ped.cs [hunk 1]
```diff
@@ -1150,6 +1281,30 @@ public void LeaveGroup()
...
+			{
+				new ArgumentOutOfRangeException("modifier");
+			}
+		}
+
+		public void PlayAmbientSpeech(string voiceName, string speechName, SpeechModifier modifier = SpeechModifier.Standard)
+		{
+			if ((int)modifier >= 0 && (int)modifier < _speechModifierNames.Length)
+			{
+				Function.Call(Hash._PLAY_AMBIENT_SPEECH1, Handle, speechName, voiceName, modifier, 0);
```
Comment (zorg93) [inline source/scripting/Ped.cs]: Shouldnt this be _PLAY_AMBIENT_SPEECH_WITH_VOICE, also this goes for this and the above native, Pretty sure that modifier is meant to be called using _speechModifierNames[modifier]

File: source/scripting/Ped.cs [hunk 2]
```diff
@@ -1150,6 +1281,30 @@ public void LeaveGroup()
...
+			{
+				new ArgumentOutOfRangeException("modifier");
+			}
+		}
+
+		public void PlayAmbientSpeech(string voiceName, string speechName, SpeechModifier modifier = SpeechModifier.Standard)
+		{
+			if ((int)modifier >= 0 && (int)modifier < _speechModifierNames.Length)
+			{
+				Function.Call(Hash._PLAY_AMBIENT_SPEECH1, Handle, speechName, voiceName, _speechModifierNames[(int)modifier], 0);
```
Comment (zorg93) [inline source/scripting/Ped.cs]: shouldnt this still be _PLAY_AMBIENT_SPEECH_WITH_VOICE

File: source/scripting/Game.cs [hunk 3]
```diff
@@ -156,19 +156,29 @@ public static float TimeScale
...
+		{
+			get
+			{
+				return 1.0f / DeltaTime;
+			}
+		}
+		/// <summary>
+		/// Gets the time it currently takes to render a frame, in seconds.
+		/// </summary>
+		public static float DeltaTime
```
Comment (crosire) [inline source/scripting/Game.cs]: What's wrong with `LastFrameTime`? Yes, `DeltaTime` means the same, but the former is more obvious.

File: source/scripting/Game.cs [hunk 4]
```diff
@@ -156,17 +156,27 @@ public static float TimeScale
...
 		}
 		/// <summary>
-		/// Gets how long(in seconds) the last frame was displayed before the next was rendered
+		/// Gets the current frame rate per second.
+		/// </summary>
+		public static float FrameRate
+		{
+			get
+			{
+				return 1.0f / DeltaTime;
```
Comment (crosire) [inline source/scripting/Game.cs]: Missed one. Also, I'd stay with `FPS` here too.

Comment (sollaholla) [inline source/scripting/Game.cs]: I don't want to budge in but maybe this should be in a `Time` class?

Comment (JohnnyCrazy) [inline source/scripting/Game.cs]: Why? The value of `FrameRate` is based on how the `Game` currently runs. Also, there are just like 2-3 props that would fit in a `Time`-class.

Comment (sollaholla) [inline source/scripting/Game.cs]: Good point. I keep thinking this is a game-engine :P

**Pull request merged**